### DISCUSSION TITLE
Nightly builds available

### DISCUSSION THREAD

**Discussion by tphakala** [tphakala/birdnet-go#330] [Announcements]
I finally managed to create a nightly build workflow for BirdNET-Go, this means that if you do not wish to use Docker build you now have pre-release updates available at https://github.com/tphakala/birdnet-go/releases

Please note that this release contains an updated TensorFlow Lite library which allows use of the XNNPACK delegate feature for accelerated AI inference. Once you have downloaded the nightly build archive, copy the included library file to your library path replacing the existing libtensorflowlite_c.so. On Windows, delete the old library which had .so extension and keep the new DLL in the same directory as birdnet-go.exe.

Also note that current AMD64 releases requires AVX2 support from CPU, this is included on Intel CPUs since Haswell generation (4th Gen Intel Core, released in 2014).

**Please backup your existing birdnet-go binary and tensorflow lite library in case you need to roll back**

After you have verified that the updated BirdNET-Go and libtensorflow_c runs on your system, you can enable XNNPACK delegate by setting usexnnpack to true under the birdnet section in config.yaml.

```yaml
birdnet:
  usexnnpack: true
```

**Comment by jkrauska** [comment]
can you comment on the performance gain?

I'm seeing what looks to be really solid performance on a rpi5.

```
2024/12/19 12:19:12 New detection: Golden-crowned Sparrow with confidence: 0.846414, source: malgo
2024/12/19 12:19:13 Updating detection: Golden-crowned Sparrow with confidence: 0.930561, source: malgo
2024/12/19 12:19:19 No species config found, using default actions for golden-crowned sparrow
2024/12/19 12:19:19 Reading segment from buffer for source: malgo
2024/12/19 12:19:39 Retrieved 5 detections in 438.792µs
2024/12/19 12:19:39 Total time for processing all notes: 598.754µs
2024/12/19 12:19:39 Spectrogram generated at 'clips/2024/12/zonotrichia_atricapilla_93p_20241219T121913Z_400px.png'
2024/12/19 12:24:13 Total time for processing all notes: 256.609µs
2024/12/19 12:24:13 Retrieved 5 detections in 447.051µs
2024/12/19 12:24:13 Retrieved 5 detections in 419.774µs
2024/12/19 12:24:13 Total time for processing all notes: 262.257µs
2024/12/19 12:24:22 Total time for processing all notes: 273.017µs
```

**Comment by uSlackr** [comment]
@tphakala running the nightly build, the version in the left navbar says unknown.  Would it be possible to add the build date there?

**Reply by uSlackr** [reply]
@tphakala Ant thoughts on this?
